<center>
    <img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/images/SN_web_lightmode.png" width="300" alt="cognitiveclass.ai logo">
</center>

<h1 align=center><font size = 5>Assignment: Notebook for Graded Assessment</font></h1>


# Introduction

Using this Python notebook you will:

1.  Understand three Chicago datasets
2.  Load the three datasets into three tables in a SQLIte database
3.  Execute SQL queries to answer assignment questions


## Understand the datasets

To complete the assignment problems in this notebook you will be using three datasets that are available on the city of Chicago's Data Portal:

1.  <a href="https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Socioeconomic Indicators in Chicago</a>
2.  <a href="https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Chicago Public Schools</a>
3.  <a href="https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Chicago Crime Data</a>

### 1. Socioeconomic Indicators in Chicago

This dataset contains a selection of six socioeconomic indicators of public health significance and a “hardship index,” for each Chicago community area, for the years 2008 – 2012.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:

[https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2](https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

### 2. Chicago Public Schools

This dataset shows all school level performance data used to create CPS School Report Cards for the 2011-2012 school year. This dataset is provided by the city of Chicago's Data Portal.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:

[https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t](https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

### 3. Chicago Crime Data

This dataset reflects reported incidents of crime (with the exception of murders where data exists for each victim) that occurred in the City of Chicago from 2001 to present, minus the most recent seven days.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:

[https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2](https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)


### Download the datasets

This assignment requires you to have these three tables populated with a subset of the whole datasets.

In many cases the dataset to be analyzed is available as a .CSV (comma separated values) file, perhaps on the internet. 

Use the links below to read the data files using the Pandas library. 

* Chicago Census Data

https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCensusData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01

* Chicago Public Schools

https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoPublicSchools.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01

* Chicago Crime Data

https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCrimeData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01

**NOTE:** Ensure you use the datasets available on the links above instead of directly from the Chicago Data Portal. The versions linked here are subsets of the original datasets and have some of the column names modified to be more database friendly which will make it easier to complete this assignment.


Execute the below code cell to install the required libraries


In [1]:
!pip install pandas
!pip install ipython-sql prettytable 

import prettytable

prettytable.DEFAULT = 'DEFAULT'

  Using cached sqlparse-0.5.3-py3-none-any.whl.metadata (3.9 kB)
   ---------------------------------------- 0.0/2.1 MB ? eta -:--:--
   ---------------------------------------- 0.0/2.1 MB ? eta -:--:--
   ---------------------------------------- 0.0/2.1 MB ? eta -:--:--
   ---- ----------------------------------- 0.3/2.1 MB ? eta -:--:--
   --------- ------------------------------ 0.5/2.1 MB 789.2 kB/s eta 0:00:03
   --------- ------------------------------ 0.5/2.1 MB 789.2 kB/s eta 0:00:03
   --------- ------------------------------ 0.5/2.1 MB 789.2 kB/s eta 0:00:03
   --------- ------------------------------ 0.5/2.1 MB 789.2 kB/s eta 0:00:03
   -------------- ------------------------- 0.8/2.1 MB 455.8 kB/s eta 0:00:03
   -------------- ------------------------- 0.8/2.1 MB 455.8 kB/s eta 0:00:03
   -------------- ------------------------- 0.8/2.1 MB 455.8 kB/s eta 0:00:03
   -------------- ------------------------- 0.8/2.1 MB 455.8 kB/s eta 0:00:03
   -------------- -----------------

### Store the datasets in database tables

To analyze the data using SQL, it first needs to be loaded into SQLite DB.
We will create three tables in as under:

1.  **CENSUS_DATA**
2.  **CHICAGO_PUBLIC_SCHOOLS**
3.  **CHICAGO_CRIME_DATA**


Load the `pandas` and `sqlite3` libraries and establish a connection to `FinalDB.db`


In [3]:
import pandas as pd
import sqlite3
conn = sqlite3.connect('FinalDB.db')
census_df = pd.read_csv('https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCensusData.csv')
public_schools_df = pd.read_csv('https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoPublicSchools.csv')
crime_df = pd.read_csv('https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCrimeData.csv')
census_df.to_sql('CENSUS_DATA', conn, if_exists='replace', index=False, method='multi')


78

In [11]:
public_schools_df.to_sql('CHICAGO_PUBLIC_SCHOOLS', conn, if_exists='replace', index=False )


566

In [17]:
crime_df.to_sql('CHICAGO_CRIME_DATA', conn, if_exists='replace', index=False, method='multi')


533

Load the SQL magic module


In [6]:
!pip install ipython-sql
%load_ext sql

In [7]:
%sql sqlite:///FinalDB.db

Use `Pandas` to load the data available in the links above to dataframes. Use these dataframes to load data on to the database `FinalDB.db` as required tables.


Establish a connection between SQL magic module and the database `FinalDB.db`


You can now proceed to the the following questions. Please note that a graded assignment will follow this lab and there will be a question on each of the problems stated below. It can be from the answer you received or the code you write for this problem. Therefore, please keep a note of both your codes as well as the response you generate.


### Store the datasets in database tables

To analyze the data using SQL, it first needs to be loaded into SQLite DB.
We will create three tables in as under:

1.  **CENSUS_DATA**
2.  **CHICAGO_PUBLIC_SCHOOLS**
3.  **CHICAGO_CRIME_DATA**

## Problems

Now write and execute SQL queries to solve assignment problems

### Problem 1

##### Find the total number of crimes recorded in the CRIME table.


In [14]:
crime_df.shape

(533, 21)

In [18]:
%sql select count(*) from CHICAGO_CRIME_DATA;

 * sqlite:///FinalDB.db
Done.


[(533,)]

### Problem 2

##### List community area names and numbers with per capita income less than 11000.


In [21]:
%sql SELECT name from sqlite_master where type='table';

 * sqlite:///FinalDB.db
Done.


[('CENSUS_DATA',), ('CHICAGO_PUBLIC_SCHOOLS',), ('CHICAGO_CRIME_DATA',)]

In [ ]:
%sql select name, type from PRAGMA_TABLE_INFO('CENSUS_DATA')

 * sqlite:///FinalDB.db
Done.


[('COMMUNITY_AREA_NUMBER', 'REAL'),
 ('COMMUNITY_AREA_NAME', 'TEXT'),
 ('PERCENT_OF_HOUSING_CROWDED', 'REAL'),
 ('PERCENT_HOUSEHOLDS_BELOW_POVERTY', 'REAL'),
 ('PERCENT_AGED_16__UNEMPLOYED', 'REAL'),
 ('PERCENT_AGED_25__WITHOUT_HIGH_SCHOOL_DIPLOMA', 'REAL'),
 ('PERCENT_AGED_UNDER_18_OR_OVER_64', 'REAL'),
 ('PER_CAPITA_INCOME', 'INTEGER'),
 ('HARDSHIP_INDEX', 'REAL')]

In [25]:
%sql select name, type from PRAGMA_TABLE_INFO('CHICAGO_CRIME_DATA');

 * sqlite:///FinalDB.db
Done.


[('ID', 'INTEGER'),
 ('CASE_NUMBER', 'TEXT'),
 ('DATE', 'TEXT'),
 ('BLOCK', 'TEXT'),
 ('IUCR', 'TEXT'),
 ('PRIMARY_TYPE', 'TEXT'),
 ('DESCRIPTION', 'TEXT'),
 ('LOCATION_DESCRIPTION', 'TEXT'),
 ('ARREST', 'INTEGER'),
 ('DOMESTIC', 'INTEGER'),
 ('BEAT', 'INTEGER'),
 ('DISTRICT', 'INTEGER'),
 ('WARD', 'REAL'),
 ('COMMUNITY_AREA_NUMBER', 'REAL'),
 ('FBICODE', 'TEXT'),
 ('X_COORDINATE', 'REAL'),
 ('Y_COORDINATE', 'REAL'),
 ('YEAR', 'INTEGER'),
 ('LATITUDE', 'REAL'),
 ('LONGITUDE', 'REAL'),
 ('LOCATION', 'TEXT')]

In [26]:
%sql select COMMUNITY_AREA_NAME, COMMUNITY_AREA_NUMBER from CENSUS_DATA where PER_CAPITA_INCOME <11000;

 * sqlite:///FinalDB.db
Done.


[('West Garfield Park', 26.0),
 ('South Lawndale', 30.0),
 ('Fuller Park', 37.0),
 ('Riverdale', 54.0)]

### Problem 3

##### List all case numbers for crimes involving minors?(children are not considered minors for the purposes of crime analysis) 


In [33]:
crime_df.head()

ID CASE_NUMBER        DATE                     BLOCK IUCR  \
0   3512276    HK587712  2004-08-28        047XX S KEDZIE AVE  890   
1   3406613    HK456306  2004-06-26  009XX N CENTRAL PARK AVE  820   
2   8002131    HT233595  2011-04-04        043XX S WABASH AVE  820   
3   7903289    HT133522  2010-12-30      083XX S KINGSTON AVE  840   
4  10402076    HZ138551  2016-02-02           033XX W 66TH ST  820   

  PRIMARY_TYPE                    DESCRIPTION          LOCATION_DESCRIPTION  \
0        THEFT                  FROM BUILDING            SMALL RETAIL STORE   
1        THEFT                 $500 AND UNDER                         OTHER   
2        THEFT                 $500 AND UNDER  NURSING HOME/RETIREMENT HOME   
3        THEFT  FINANCIAL ID THEFT: OVER $300                     RESIDENCE   
4        THEFT                 $500 AND UNDER                         ALLEY   

   ARREST  DOMESTIC  ...  DISTRICT  WARD  COMMUNITY_AREA_NUMBER  FBICODE  \
0   False     False  ...         9  14.0                   58.0        6   
1   False     False  ...        11  27.0                   23.0        6   
2   False     False  ...         2   3.0                   38.0        6   
3   False     False  ...         4   7.0                   46.0        6   
4   False     False  ...         8  15.0                   66.0        6   

  X_COORDINATE  Y_COORDINATE  YEAR   LATITUDE  LONGITUDE  \
0    1155838.0     1873050.0  2004  41.807440 -87.703956   
1    1152206.0     1906127.0  2004  41.898280 -87.716406   
2    1177436.0     1876313.0  2011  41.815933 -87.624642   
3    1194622.0     1850125.0  2010  41.743665 -87.562463   
4    1155240.0     1860661.0  2016  41.773455 -87.706480   

                        LOCATION  
0    (41.8074405, -87.703955849)  
1  (41.898279962, -87.716405505)  
2  (41.815933131, -87.624642127)  
3  (41.743665322, -87.562462756)  
4  (41.773455295, -87.706480471)  

[5 rows x 21 columns]

In [82]:
%sql select "CASE_NUMBER","PRIMARY_TYPE", "DESCRIPTION" from CHICAGO_CRIME_DATA where "PRIMARY_TYPE" like '%chil%';


 * sqlite:///FinalDB.db
Done.


[('HN567387', 'OFFENSE INVOLVING CHILDREN', 'AGG SEX ASSLT OF CHILD FAM MBR'),
 ('HR391350', 'OFFENSE INVOLVING CHILDREN', 'SEX ASSLT OF CHILD BY FAM MBR'),
 ('HM768251', 'OFFENSE INVOLVING CHILDREN', 'CRIM SEX ABUSE BY FAM MEMBER'),
 ('HT394616', 'OFFENSE INVOLVING CHILDREN', 'AGG CRIM SEX ABUSE FAM MEMBER')]

### Problem 4

##### List all kidnapping crimes involving a child?


In [35]:
%sql select * from CHICAGO_CRIME_DATA where "PRIMARY_TYPE" like '%kidnap%';

 * sqlite:///FinalDB.db
Done.


[(5276766, 'HN144152', '2007-01-26', '050XX W VAN BUREN ST', '1792', 'KIDNAPPING', 'CHILD ABDUCTION/STRANGER', 'STREET', 0, 0, 1533, 15, 29.0, 25.0, '20', 1143050.0, 1897546.0, 2007, 41.87490841, -87.75024931, '(41.874908413, -87.750249307)')]

### Problem 5

##### List the kind of crimes that were recorded at schools. (No repetitions)


In [36]:
%sql select distinct "PRIMARY_TYPE" from CHICAGO_CRIME_DATA where "LOCATION_DESCRIPTION" like '%school%';

 * sqlite:///FinalDB.db
Done.


[('BATTERY',),
 ('CRIMINAL DAMAGE',),
 ('NARCOTICS',),
 ('ASSAULT',),
 ('CRIMINAL TRESPASS',),
 ('PUBLIC PEACE VIOLATION',)]

In [43]:
%sql SELECT name, type from PRAGMA_TABLE_INFO('CHICAGO_PUBLIC_SCHOOLS');

 * sqlite:///FinalDB.db
Done.


[('School_ID', 'INTEGER'),
 ('NAME_OF_SCHOOL', 'TEXT'),
 ('Elementary, Middle, or High School', 'TEXT'),
 ('Street_Address', 'TEXT'),
 ('City', 'TEXT'),
 ('State', 'TEXT'),
 ('ZIP_Code', 'INTEGER'),
 ('Phone_Number', 'TEXT'),
 ('Link', 'TEXT'),
 ('Network_Manager', 'TEXT'),
 ('Collaborative_Name', 'TEXT'),
 ('Adequate_Yearly_Progress_Made_', 'TEXT'),
 ('Track_Schedule', 'TEXT'),
 ('CPS_Performance_Policy_Status', 'TEXT'),
 ('CPS_Performance_Policy_Level', 'TEXT'),
 ('HEALTHY_SCHOOL_CERTIFIED', 'TEXT'),
 ('Safety_Icon', 'TEXT'),
 ('SAFETY_SCORE', 'REAL'),
 ('Family_Involvement_Icon', 'TEXT'),
 ('Family_Involvement_Score', 'TEXT'),
 ('Environment_Icon', 'TEXT'),
 ('Environment_Score', 'REAL'),
 ('Instruction_Icon', 'TEXT'),
 ('Instruction_Score', 'REAL'),
 ('Leaders_Icon', 'TEXT'),
 ('Leaders_Score', 'TEXT'),
 ('Teachers_Icon', 'TEXT'),
 ('Teachers_Score', 'TEXT'),
 ('Parent_Engagement_Icon', 'TEXT'),
 ('Parent_Engagement_Score', 'TEXT'),
 ('Parent_Environment_Icon', 'TEXT'),
 ('Parent_Environment_Score', 'TEXT'),
 ('AVERAGE_STUDENT_ATTENDANCE', 'TEXT'),
 ('Rate_of_Misconducts__per_100_students_', 'REAL'),
 ('Average_Teacher_Attendance', 'TEXT'),
 ('Individualized_Education_Program_Compliance_Rate', 'TEXT'),
 ('Pk_2_Literacy__', 'TEXT'),
 ('Pk_2_Math__', 'TEXT'),
 ('Gr3_5_Grade_Level_Math__', 'TEXT'),
 ('Gr3_5_Grade_Level_Read__', 'TEXT'),
 ('Gr3_5_Keep_Pace_Read__', 'TEXT'),
 ('Gr3_5_Keep_Pace_Math__', 'TEXT'),
 ('Gr6_8_Grade_Level_Math__', 'TEXT'),
 ('Gr6_8_Grade_Level_Read__', 'TEXT'),
 ('Gr6_8_Keep_Pace_Math_', 'TEXT'),
 ('Gr6_8_Keep_Pace_Read__', 'TEXT'),
 ('Gr_8_Explore_Math__', 'TEXT'),
 ('Gr_8_Explore_Read__', 'TEXT'),
 ('ISAT_Exceeding_Math__', 'REAL'),
 ('ISAT_Exceeding_Reading__', 'REAL'),
 ('ISAT_Value_Add_Math', 'REAL'),
 ('ISAT_Value_Add_Read', 'REAL'),
 ('ISAT_Value_Add_Color_Math', 'TEXT'),
 ('ISAT_Value_Add_Color_Read', 'TEXT'),
 ('Students_Taking__Algebra__', 'TEXT'),
 ('Students_Passing__Algebra__', 'TEXT'),
 ('9th Grade EXPLORE (2009)', 'TEXT'),
 ('9th Grade EXPLORE (2010)', 'TEXT'),
 ('10th Grade PLAN (2009)', 'TEXT'),
 ('10th Grade PLAN (2010)', 'TEXT'),
 ('Net_Change_EXPLORE_and_PLAN', 'TEXT'),
 ('11th Grade Average ACT (2011)', 'TEXT'),
 ('Net_Change_PLAN_and_ACT', 'TEXT'),
 ('College_Eligibility__', 'TEXT'),
 ('Graduation_Rate__', 'TEXT'),
 ('College_Enrollment_Rate__', 'TEXT'),
 ('COLLEGE_ENROLLMENT', 'INTEGER'),
 ('General_Services_Route', 'INTEGER'),
 ('Freshman_on_Track_Rate__', 'TEXT'),
 ('X_COORDINATE', 'REAL'),
 ('Y_COORDINATE', 'REAL'),
 ('Latitude', 'REAL'),
 ('Longitude', 'REAL'),
 ('COMMUNITY_AREA_NUMBER', 'INTEGER'),
 ('COMMUNITY_AREA_NAME', 'TEXT'),
 ('Ward', 'INTEGER'),
 ('Police_District', 'INTEGER'),
 ('Location', 'TEXT')]

In [37]:
public_schools_df.head()

School_ID                                     NAME_OF_SCHOOL  \
0     610038                  Abraham Lincoln Elementary School   
1     610281  Adam Clayton Powell Paideia Community Academy ...   
2     610185                Adlai E Stevenson Elementary School   
3     609993                    Agustin Lara Elementary Academy   
4     610513                      Air Force Academy High School   

  Elementary, Middle, or High School         Street_Address     City State  \
0                                 ES        615 W Kemper Pl  Chicago    IL   
1                                 ES  7511 S South Shore Dr  Chicago    IL   
2                                 ES     8010 S Kostner Ave  Chicago    IL   
3                                 ES     4619 S Wolcott Ave  Chicago    IL   
4                                 HS        3630 S Wells St  Chicago    IL   

   ZIP_Code    Phone_Number  \
0     60614  (773) 534-5720   
1     60649  (773) 535-6650   
2     60652  (773) 535-2280   
3     60609  (773) 535-4389   
4     60609  (773) 535-1590   

                                                Link  \
0  http://schoolreports.cps.edu/SchoolProgressRep...   
1  http://schoolreports.cps.edu/SchoolProgressRep...   
2  http://schoolreports.cps.edu/SchoolProgressRep...   
3  http://schoolreports.cps.edu/SchoolProgressRep...   
4  http://schoolreports.cps.edu/SchoolProgressRep...   

                      Network_Manager  ... Freshman_on_Track_Rate__  \
0        Fullerton Elementary Network  ...                      NDA   
1           Skyway Elementary Network  ...                      NDA   
2           Midway Elementary Network  ...                      NDA   
3         Pershing Elementary Network  ...                      NDA   
4  Southwest Side High School Network  ...                     91.8   

  X_COORDINATE Y_COORDINATE   Latitude  Longitude COMMUNITY_AREA_NUMBER  \
0  1171699.458  1915829.428  41.924497 -87.644522                     7   
1  1196129.985  1856209.466  41.760324 -87.556736                    43   
2  1148427.165  1851012.215  41.747111 -87.731702                    70   
3  1164504.290  1873959.199  41.809757 -87.672145                    61   
4  1175177.622  1880745.126  41.828146 -87.632794                    34   

  COMMUNITY_AREA_NAME  Ward Police_District                     Location  
0        LINCOLN PARK    43              18  (41.92449696, -87.64452163)  
1         SOUTH SHORE     7               4  (41.76032435, -87.55673627)  
2             ASHBURN    13               8  (41.74711093, -87.73170248)  
3            NEW CITY    20               9    (41.8097569, -87.6721446)  
4       ARMOUR SQUARE    11               9  (41.82814609, -87.63279369)  

[5 rows x 78 columns]

### Problem 6

##### List the type of schools along with the average safety score for each type.


In [44]:
%sql select `Elementary, Middle, or High School`, AVG("SAFETY_SCORE") as avg_safety_score from CHICAGO_PUBLIC_SCHOOLS group by `Elementary, Middle, or High School`;

 * sqlite:///FinalDB.db
Done.


[('ES', 49.52038369304557), ('HS', 49.62352941176471), ('MS', 48.0)]

In [45]:
census_df.head()

COMMUNITY_AREA_NUMBER COMMUNITY_AREA_NAME  PERCENT_OF_HOUSING_CROWDED  \
0                    1.0         Rogers Park                         7.7   
1                    2.0          West Ridge                         7.8   
2                    3.0              Uptown                         3.8   
3                    4.0      Lincoln Square                         3.4   
4                    5.0        North Center                         0.3   

   PERCENT_HOUSEHOLDS_BELOW_POVERTY  PERCENT_AGED_16__UNEMPLOYED  \
0                              23.6                          8.7   
1                              17.2                          8.8   
2                              24.0                          8.9   
3                              10.9                          8.2   
4                               7.5                          5.2   

   PERCENT_AGED_25__WITHOUT_HIGH_SCHOOL_DIPLOMA  \
0                                          18.2   
1                                          20.8   
2                                          11.8   
3                                          13.4   
4                                           4.5   

   PERCENT_AGED_UNDER_18_OR_OVER_64  PER_CAPITA_INCOME  HARDSHIP_INDEX  
0                              27.5              23939            39.0  
1                              38.5              23040            46.0  
2                              22.2              35787            20.0  
3                              25.5              37524            17.0  
4                              26.2              57123             6.0

### Problem 7

##### List 5 community areas with highest % of households below poverty line


In [47]:
%sql SELECT "COMMUNITY_AREA_NAME" FROM CENSUS_DATA  order by "PERCENT_HOUSEHOLDS_BELOW_POVERTY" DESC LIMIT 5;

 * sqlite:///FinalDB.db
Done.


[('Riverdale',),
 ('Fuller Park',),
 ('Englewood',),
 ('North Lawndale',),
 ('East Garfield Park',)]

In [51]:
crime_df.columns

Index(['ID', 'CASE_NUMBER', 'DATE', 'BLOCK', 'IUCR', 'PRIMARY_TYPE',
       'DESCRIPTION', 'LOCATION_DESCRIPTION', 'ARREST', 'DOMESTIC', 'BEAT',
       'DISTRICT', 'WARD', 'COMMUNITY_AREA_NUMBER', 'FBICODE', 'X_COORDINATE',
       'Y_COORDINATE', 'YEAR', 'LATITUDE', 'LONGITUDE', 'LOCATION'],
      dtype='object')

### Problem 8

##### Which community area is most crime prone? Display the coumminty area number only.



In [79]:
%sql SELECT "COMMUNITY_AREA_NUMBER", count(*) as "crime_count" FROM CHICAGO_CRIME_DATA GROUP BY "COMMUNITY_AREA_NUMBER" HAVING count(*) = (SELECT MAX(crime_count) FROM (SELECT count(*) as crime_count FROM CHICAGO_CRIME_DATA GROUP BY "COMMUNITY_AREA_NUMBER"));

 * sqlite:///FinalDB.db
Done.


[(None, 43), (25.0, 43)]


Double-click **here** for a hint

<!--
Query for the 'community area number' that has most number of incidents
-->


In [78]:
census_df.columns

Index(['COMMUNITY_AREA_NUMBER', 'COMMUNITY_AREA_NAME',
       'PERCENT_OF_HOUSING_CROWDED', 'PERCENT_HOUSEHOLDS_BELOW_POVERTY',
       'PERCENT_AGED_16__UNEMPLOYED',
       'PERCENT_AGED_25__WITHOUT_HIGH_SCHOOL_DIPLOMA',
       'PERCENT_AGED_UNDER_18_OR_OVER_64', 'PER_CAPITA_INCOME',
       'HARDSHIP_INDEX'],
      dtype='object')

### Problem 9

##### Use a sub-query to find the name of the community area with highest hardship index


In [62]:
%sql SELECT "COMMUNITY_AREA_NAME" FROM CENSUS_DATA WHERE "HARDSHIP_INDEX" = (select max("HARDSHIP_INDEX") from CENSUS_DATA);

 * sqlite:///FinalDB.db
Done.


[('Riverdale',)]

### Problem 10

##### Use a sub-query to determine the Community Area Name with most number of crimes?


In [65]:
%sql select * FROM CHICAGO_CRIME_DATA where "COMMUNITY_AREA_NUMBER" = 25.0 limit 5;

 * sqlite:///FinalDB.db
Done.


[(9166453, 'HW303994', '2013-06-04', '055XX W GLADYS AVE', '810', 'THEFT', 'OVER $500', 'VACANT LOT/LAND', 0, 0, 1522, 15, 29.0, 25.0, '6', 1139693.0, 1897797.0, 2013, 41.87565911, -87.76256886, '(41.875659108, -87.762568861)'),
 (7762423, 'HS569246', '2010-10-17', '058XX W ARTHINGTON ST', '880', 'THEFT', 'PURSE-SNATCHING', 'SIDEWALK', 0, 0, 1513, 15, 29.0, 25.0, '6', 1137635.0, 1895420.0, 2010, 41.8691736, -87.77018245, '(41.869173601, -87.770182447)'),
 (2688426, 'HJ308632', '2003-04-17', '017XX N AUSTIN AVE', '810', 'THEFT', 'OVER $500', 'RESIDENCE-GARAGE', 0, 0, 2531, 25, 29.0, 25.0, '6', 1136138.0, 1910943.0, 2003, 41.91179756, -87.77530779, '(41.911797557, -87.775307788)'),
 (10943516, 'JA261898', '2017-05-13', '0000X N LATROBE AVE', '820', 'THEFT', '$500 AND UNDER', 'STREET', 0, 0, 1522, 15, 28.0, 25.0, '6', 1141381.0, 1899903.0, 2017, 41.88140727, -87.75631908, '(41.88140727, -87.756319082)'),
 (10485657, 'HZ224258', '2016-04-12', '017XX N LUNA AVE', '820', 'THEFT', '$500 AND UNDER', 'STREET', 0, 0, 2532, 25, 37.0, 25.0, '6', 1139131.0, 1911034.0, 2016, 41.91199331, -87.76431002, '(41.911993311, -87.764310023)')]

In [69]:
%sql SELECT "COMMUNITY_AREA_NAME" FROM CENSUS_DATA WHERE "COMMUNITY_AREA_NUMBER" = (SELECT "COMMUNITY_AREA_NUMBER" FROM CHICAGO_CRIME_DATA where "COMMUNITY_AREA_NUMBER" = (select MAX("COMMUNITY_AREA_NUMBER") from CHICAGO_CRIME_DATA));

 * sqlite:///FinalDB.db
Done.


[('Edgewater',)]

## Author(s)

<h4> Hima Vasudevan </h4>
<h4> Rav Ahuja </h4>
<h4> Ramesh Sannreddy </h4>

## Contribtuor(s)

<h4> Malika Singla </h4>
<h4>Abhishek Gagneja</h4>
<!--
## Change log

| Date       | Version | Changed by        | Change Description                             |
| ---------- | ------- | ----------------- | ---------------------------------------------- |
|2023-10-18  | 2.6     | Abhishek Gagneja  | Modified instruction set |
| 2022-03-04 | 2.5     | Lakshmi Holla     | Changed markdown.                   |
| 2021-05-19 | 2.4     | Lakshmi Holla     | Updated the question                           |
| 2021-04-30 | 2.3     | Malika Singla     | Updated the libraries                          |
| 2021-01-15 | 2.2     | Rav Ahuja         | Removed problem 11 and fixed changelog         |
| 2020-11-25 | 2.1     | Ramesh Sannareddy | Updated the problem statements, and datasets   |
| 2020-09-05 | 2.0     | Malika Singla     | Moved lab to course repo in GitLab             |
| 2018-07-18 | 1.0     | Rav Ahuja         | Several updates including loading instructions |
| 2018-05-04 | 0.1     | Hima Vasudevan    | Created initial version                        |
-->
## <h3 align="center"> © IBM Corporation 2023. All rights reserved. <h3/>
